pyquist/audio.py

```diff
@@ -95,14 +95,46 @@ def __new__(
         obj._sample_rate = sample_rate
         return obj
 
-    def __getitem__(self, *args, **kwargs):
-        """Ensure slices retain the sample rate."""
-        result = super().__getitem__(*args, **kwargs)
-        if isinstance(result, AudioBuffer):
-            result._sample_rate = self._sample_rate
-            assert isinstance(result, Audio)
```

Comment:
Handled in more generic __array_function__ function below

Reply:
nice!